crates/runtime/src/memory.rs

```diff
@@ -432,7 +433,7 @@ impl RuntimeLinearMemory for StaticMemory {
 /// [thread proposal]:
 ///     https://github.com/WebAssembly/threads/blob/master/proposals/threads/Overview.md#webassemblymemoryprototypegrow
 #[derive(Clone)]
-pub struct SharedMemory(Arc<RwLock<SharedMemoryInner>>);
+pub struct SharedMemory(Arc<RwLock<SharedMemoryInner>>, Arc<ParkingSpot>);
```

Comment:
I think it would be better to put the `ParkingSpot` adjacent to the `RwLock` in memory to avoid allocating two `Arc`s here. Additionally as this struct grows can you use named fields rather than tuple structs?